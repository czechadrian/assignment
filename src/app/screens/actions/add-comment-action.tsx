import { ActionType, createAction } from "typesafe-actions";
import { TComment } from "../../reducers/comments";

export const addCommentAction = createAction("comment/ADD_NEW_COMMENT")<
  TComment
>();

export type TAddCommentAction = ActionType<typeof addCommentAction>;
