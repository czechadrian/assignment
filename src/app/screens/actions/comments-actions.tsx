import { ActionType, createAction } from "typesafe-actions";
import { ThunkAction } from "redux-thunk";
import { TRootState } from "../../reducers";
import { getComments } from "../../../api-wrapper/get-comments";
import { TComments } from "../../reducers/comments";

export const getCommentsInitAction = createAction("data/FETCH_INIT")();
export const getCommentsSuccessAction = createAction("data/FETCH_SUCCESS")<
  Pick<TComments, "comments">
>();
export const getCommentsFailureAction = createAction("data/FETCH_FAILURE")();

export type TGetCommentsInitAction = ActionType<typeof getCommentsInitAction>;
export type TGetCommentsSuccessAction = ActionType<
  typeof getCommentsSuccessAction
>;
export type TGetCommentsFailureAction = ActionType<
  typeof getCommentsFailureAction
>;

export type TGetCommentsActions =
  | TGetCommentsSuccessAction
  | TGetCommentsFailureAction
  | TGetCommentsInitAction;

export type TFetchCommentsAction = () => ThunkAction<
  void,
  TRootState,
  null,
  TGetCommentsActions
>;
export const fetchComments: TFetchCommentsAction = () => (dispatch) => {
  dispatch(getCommentsInitAction());
  return getComments()
    .then((payload) => {
      return dispatch(getCommentsSuccessAction({ comments: payload }));
    })
    .catch(() => {
      dispatch(getCommentsFailureAction());
    });
};
