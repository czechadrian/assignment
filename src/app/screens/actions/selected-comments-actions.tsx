import { ActionType, createAction } from "typesafe-actions";
import { TComment } from "../../reducers/comments";

export const selectCommentAction = createAction(
  "homepage/CLICK_SELECT_COMMENT"
)<Pick<TComment, "id">>();
export const removeSelectedCommentAction = createAction(
  "selected-comments/REMOVE_SELECTED_COMMENT"
)<Pick<TComment, "id">>();

export type TSelectCommentAction = ActionType<typeof selectCommentAction>;

export type TRemoveSelectedCommentAction = ActionType<
  typeof removeSelectedCommentAction
>;

export type TSelectedCommentActions =
  | TSelectCommentAction
  | TRemoveSelectedCommentAction;
