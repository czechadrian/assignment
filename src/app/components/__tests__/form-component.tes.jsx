import React from "react";
import { useDispatch, useSelector } from "react-redux";
import "jest-styled-components";
import renderer from "react-test-renderer";
import CustomForm from "../form-component";
import { addCommentAction } from "../../screens/actions/add-comment-action";

jest.mock("../../screens/actions/add-comment-action", () => {
  return {
    addCommentAction: jest.fn(),
  };
});

jest.mock("react-redux", () => {
  return {
    useSelector: jest.fn(),
    useDispatch: jest.fn(),
  };
});

describe("CustomForm", () => {
  const dispathMock = jest.fn();
  const selectorMock = jest.fn();
  beforeEach(() => {
    useSelector.mockImplementation(selectorMock);
    useDispatch.mockReturnValue(dispathMock);

    addCommentAction.mockReturnValue(() => jest.fn());

    jest.clearAllMocks();
  });
  test("selected comments component test", () => {
    const tree = renderer.create(<CustomForm />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
