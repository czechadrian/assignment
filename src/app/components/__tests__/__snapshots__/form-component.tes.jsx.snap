// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomForm selected comments component test 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  line-height: 2;
  text-align: left;
  display: block;
  margin-bottom: 13px;
  margin-top: 20px;
  color: black;
  font-size: 14px;
  font-weight: 200;
}

.c3 {
  background: #2196f3;
  color: white;
  text-transform: uppercase;
  border: none;
  margin-top: 40px;
  padding: 20px;
  font-size: 16px;
  font-weight: 100;
  -webkit-letter-spacing: 10px;
  -moz-letter-spacing: 10px;
  -ms-letter-spacing: 10px;
  letter-spacing: 10px;
}

.c1 {
  width: 500px;
}

<form
  className="c0"
  onSubmit={[Function]}
>
  <div
    className="c1"
  >
    <label
      className="c2"
    >
      Name
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline controller"
      onClick={[Function]}
    >
      <input
        className="MuiInputBase-input MuiInput-input"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="c1"
  >
    <label
      className="c2"
    >
      Email
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline controller"
      onClick={[Function]}
    >
      <input
        className="MuiInputBase-input MuiInput-input"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="c1"
  >
    <label
      className="c2"
    >
      Text
    </label>
    <div
      className="MuiInputBase-root MuiInput-root MuiInput-underline controller"
      onClick={[Function]}
    >
      <input
        className="MuiInputBase-input MuiInput-input"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
    </div>
  </div>
  <input
    className="c3"
    type="submit"
    value="Add New Comment"
  />
</form>
`;
